### PR TITLE
Add Object>>traceCr:, correct ADR 0010 OTP placement, update examples (BT-984)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Adds `Object>>trace:` and `Object>>traceCr:` methods following the Pharo/GNU Smalltalk convention. These are nil-safe — they delegate to `TranscriptStream current` via `ifNotNil:`, making them no-ops in batch compile or test contexts.

- Add `trace:` (no newline) and `traceCr:` (with newline) to `Object.bt`
- Add BUnit tests covering nil no-op and output routing when transcript is set
- Update examples (`hanoi.bt`, `hello.bt`, `stream.bt`, `gof main.bt`) to use `traceCr:`/`trace:` instead of direct `TranscriptStream current` sends
- Correct ADR 0010 OTP placement: split TranscriptStream into class definition (runtime) vs singleton instance (workspace)

## Linear Issue

https://linear.app/beamtalk/issue/BT-984

## Test plan

- [x] `just test-stdlib` passes (237/237)
- [x] `just ci` passes (pre-existing flaky `test_stop_workspace_force_integration` excluded)
- [x] BUnit tests verify nil-safe no-op and transcript routing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added trace: and traceCr: methods on Object to emit strings to the transcript stream; calls are nil-safe when no transcript is active.

* **Documentation**
  * Updated architecture docs to clarify TranscriptStream vs. the workspace-bound Transcript singleton and their startup/compile semantics.

* **Tests**
  * Added unit tests covering nil-safe behavior and routing to an active transcript.

### CODE REVIEW

File: stdlib/src/Object.bt [hunk 1]
```diff
@@ -216,6 +216,26 @@ ProtoObject subclass: Object
...
+
+  /// Send aString to the current transcript followed by a newline.
+  ///
+  /// Nil-safe: does nothing when no transcript is set (batch compile, tests).
+  ///
+  /// ## Examples
+  /// ```beamtalk
+  /// 42 traceCr: "hello world"
+  /// ```
+  traceCr: aString: String => TranscriptStream current ifNotNil: [:t | t show: aString; cr]
```
[the remark this answers is not in the document]
Comment (jamesc) [reply stdlib/src/Object.bt]: Fixed alongside `trace:` — both now accept any `Object`. See e1ce82f5.
